SCons/Environment.py

```diff
@@ -2424,6 +2433,14 @@ def values(self):
         """Emulates the values() method of dictionaries."""
         return self.Dictionary().values()
 
+    def setdefault(self, key, default=None):
+        """Emulates the setdefault() method of dictionaries."""
+        try:
+            return self.Dictionary(key)
```

Comment:
Won't this force an unneeded copy of parent's Dictionary() ?

Reply:
if forces a copy, but that copy/update is a straightforward way to fetch the key's value from whichever dict it appears in (or take KeyError if neither). That's not to say this could not be implemented a different way as well, using a double try.

Reply:
use .get() ?

Reply:
I can use `__getattr__` ... `get` isn't as useful, want to hear it if `key` doesn't exist.

Reply:
`__getitem__` was a better choice.